### PR TITLE
Government/Innovations/Holdings implementation

### PR DESCRIPTION
**Pull request by LT-Rascek**
First cut of implementing innovations, holdings, and government/laws/vassal contracts.

* Innovations are more precisely mapped out if not fleshed out.
* Default holdings are going to need to be rewritten so they were ported in.
* Added the new culture eras and some placeholder effects (e.g., haven't touch the actual development math yet)
* Some debug decisions for testing in CK3

Still very early implementation that will need a lot more work to complete. Unfortunately, innovations is really closely tied to all these mechanics, so its easier to work on them together than separately.

### CODE REVIEW

**Comment by LT-Rascek** [comment]
Tested adding foederati as a special contract for tribals; a number of default triggers (that prevent renegotiating contracts for republics, tribals, and theocracies) prevents sending contract updates even when foederati checkbox is active in the interaction window. I suggest considering making foederati a special government type instead so we can sidestep those triggers and avoid touching too much vanilla code.

**Comment by loup99** [comment]
> I suggest considering making foederati a special government type instead so we can sidestep those triggers and avoid touching too much vanilla code.

The foederati relationship is that of a diplomatic contract, not a way of governing. I don't see how it could be a government type given that no governance is implied here.

**Comment by LT-Rascek** [comment]
> The foederati relationship is that of a diplomatic contract, not a way of governing. I don't see how it could be a government type given that no governance is implied here.

I agree it is not strictly a government type; I am noting, however, that implementing that specific contract in tribal governments will possibly involve touching far more of the vanilla code than implementing a new government. The danger with that is the strong potential for more bugs being introduced or breaking unrelated aspects. Regardless, I'll look at implementing foederatus contracts as a special contract for tribal vassals; it should also have a character interaction for bureaucratic/subromans with tribals that aren't already vassals.

@loup99 : As I've been working on this, here's some items that have jumped at me that'd require some feedback from you:
* For Imperial Elective, who should be electors/candidates and how should be weight them? Along with importing the co-emperor option from Imperial Government Mod?
* Do we want to add additional law types (like centralization, religious laws) to this mod, as was the case in WtWSMS? It's quite possible, but the GUI for the realm would need to be refactored to support it.
* How many additional holding levels/building levels do we want? I've planned on two for non-tribal holdings and one for buildings (level 0), along with staggering level 1 and 2 buildings across the early eras.
* Nature of the different government types? Biggest interest is what obligations do vassal of those governments have? I've stamped out some of them, but if you had other ideas I'd like to hear it.

**Comment by AvalonXD** [comment]
> The foederati relationship is that of a diplomatic contract, not a way of governing. I don't see how it could be a government type given that no governance is implied here.

> I agree it is not strictly a government type; I am noting, however, that implementing that specific contract in tribal governments will possibly involve touching far more of the vanilla code than implementing a new government.

While we shouldn't code with uncertain future events in mind they almost certainly will be adding tributaries and various such contracts and relationship when they get around to the Byzantine update.

**Comment by loup99** [comment]
> For Imperial Elective, who should be electors/candidates and how should be weight them? 

In CK2 we used the various minor titles, but those are no longer a thing in CK3. I think with this in mind it should be similar to vanilla CK2: https://ck2.paradoxwikis.com/Imperial_elective#Mechanics

> Along with importing the co-emperor option from Imperial Government Mod?

The way it works represents later changes, so I don't think it fits for the timeframe.

> Do we want to add additional law types (like centralization, religious laws) to this mod, as was the case in WtWSMS? It's quite possible, but the GUI for the realm would need to be refactored to support it.

Centralisation was from vanilla. We had added the separate laws for religious tolerance (although "tolerance" is anachronistic), which could still be pertinent, yes.

> How many additional holding levels/building levels do we want? I've planned on two for non-tribal holdings and one for buildings (level 0), along with staggering level 1 and 2 buildings across the early eras.

Doesn't that depend on technology ?

> Nature of the different government types? Biggest interest is what obligations do vassal of those governments have? I've stamped out some of them, but if you had other ideas I'd like to hear it.

Well, bureaucratic vassals should provide few to no levies and possible to revoke easily, but instead it is easier to maintain professional soldiers and you get more taxes.

Could this picture be of use?
Governments

**Comment by LT-Rascek** [comment]
> While we shouldn't code with uncertain future events in mind they almost certainly will be adding tributaries and various such contracts and relationship when they get around to the Byzantine update.

I agree they'll probably do that with a relatively early update, but it's not clear *when* they will do that. Until then, we have a small hole between BP and WtWSMS, as there are several extant foederati/tributary relationships in 476.

> In CK2 we used the various minor titles, but those are no longer a thing in CK3. I think with this in mind it should be similar to vanilla CK2: https://ck2.paradoxwikis.com/Imperial_elective#Mechanics

> The way it works represents later changes in the 7th and 9th centuries, so I don't think it fits for the timeframe.

Will do.

> Centralisation was from vanilla. We had added the separate laws for religious tolerance (although "tolerance" is anachronistic), which could still be pertinent, yes.

While it could be pertinent, probably best to keep things as close to vanilla as possible for alpha v0.1, then branch out from there.

> Doesn't that depend on the innovation, with the eras, we already discussed? I think that if we add new eras as stated previously the governments will have to reflect that.

That's fine; I'll try and update Issue #5 to fully reflect this (hopefully this week).

> Well, bureaucratic vassals should provide few to no levies and possible to revoke easily, but instead it is easier to maintain professional soldiers and you get more taxes.
Could this picture be of use?

It's somewhat useful; I was looking more for specific values and such for vassal levies, &c. Now that I think about it, since those values might be a moving target, I'll pivot instead to stamping out governments singly moving forward and try to give a greater breadth of detail so it is clear how it is **modeled on WtWSMS and where it deviates (and for what reasons)**. 

[AN: Bold added for emphasis]

The biggest concern (with this commit at this time) is the fact that vassal obligations/contracts are defined at the vassal level, not the liege level. That's what caused issues with my first attempt at implementing foederatus contracts, but it is not a true blocker; example partial workarounds are listed in common\vassal_contracts\bureacratic.txt right now for exarchate/theme contracts.

> **bureaucratic vassals**

This jumped out at me; there's no bureaucratic vassal obligations in this commit, because bureaucratic vassals didn't (to my knowledge) exist in WtWSMS, so common\vassal_contracts\bureacratic.txt instead contains the special contracts for vassals of **bureaucratic lieges**. The governmental requirements for bureaucratic similarly only allow independent empires to be bureaucratic.

If we wanted to pivot bureaucratic to represent the Diocletian/Constantine separation of military and civilian offices and have other governments (military command) model the post-7th changes to the Byzantine state, that's not an issue. I'll leave it to you @loup99 to decide on how things should be done. Or we can keep the WtWSMS model and make the vassals of a bureaucratic government be subroman.

For me, I'm going to focus on the less complicated governments like high tribal and prefeudal first so as to complete it with notes on how they are modeled on WtWSMS and where they deviate (and for what reasons). I think doing that will make it easier to have a conversation about how to model various aspects in CK3.

**Comment by loup99** [comment]
> I agree they'll probably do that with a relatively early update, but it's not clear when they will do that. Until then, we have a small hole between BP and WtWSMS, as there are several extant foederati/tributary relationships in 476.

I will second you on this.

> While it could be pertinent, probably best to keep things as close to vanilla as possible for alpha v0.1, then branch out from there.

Agreed, I don't think this is a priority at the moment.

> That's fine; I'll try and update Issue #5 to fully reflect this (hopefully this week).

No problems, take all the time you need.

> It's somewhat useful; I was looking more for specific values and such for vassal levies, &c. Now that I think about it, since those values might be a moving target, I'll pivot instead to stamping out governments singly moving forward and try to give a greater breadth of detail so it is clear how it is modeled on WtWSMS and where it deviates (and for what reasons).
> 
> [AN: Bold added for emphasis]
> 
> The biggest concern (with this commit at this time) is the fact that vassal obligations/contracts are defined at the vassal level, not the liege level. That's what caused issues with my first attempt at implementing foederatus contracts, but it is not a true blocker; example partial workarounds are listed in common\vassal_contracts\bureacratic.txt right now for exarchate/theme contracts.
> 
> bureaucratic vassals
> 
> This jumped out at me; there's no bureaucratic vassal obligations in this commit, because bureaucratic vassals didn't (to my knowledge) exist in WtWSMS, so common\vassal_contracts\bureacratic.txt instead contains the special contracts for vassals of bureaucratic lieges. The governmental requirements for bureaucratic similarly only allow independent empires to be bureaucratic.
> 
> If we wanted to pivot bureaucratic to represent the Diocletian/Constantine separation of military and civilian offices and have other governments (military command) model the post-7th changes to the Byzantine state, that's not an issue. I'll leave it to you @loup99 to decide on how things should be done. Or we can keep the WtWSMS model and make the vassals of a bureaucratic government be subroman.
> 

My bad, I was commenting about how it worked in CK2, hence why I said "bureaucratic vassals" because in CK2 the Bureaucratic vassal type of the liege determines the levies and taxation the vassals provide. My take as of writing would be to keep the existing setup (which I conceived to some extent with the rest of the team), although I'm open to revise it if other points of view are expressed.

> For me, I'm going to focus on the less complicated governments like high tribal and prefeudal first so as to complete it with notes on how they are modeled on WtWSMS and where they deviate (and for what reasons). I think doing that will make it easier to have a conversation about how to model various aspects in CK3.

Sure, sounds good.

**Comment by LT-Rascek** [comment]
b4e3e75 has the sketch of how the buildings and innovations are interacting, provisionally. I should have gotten the building sequence dependence (i.e., quaries_00 -> quaries_01 &c.) but I might have missed some. I'll update #5 with newer data to reflect b4e3e75 later.

Regional/cultural innovations need to be completed, governments are still incomplete (some of the authorities haven't been really finished yet), nor have I gotten the kinks worked out of the imperial elective law, nor have I done any real work on Eranshar or Gupta governments. But this was in some ways the largest single impediment on this effort, since it really only made sense to do the buildings as a single block (for simplicity if nothing else). 

Also rebased this branch onto the latest master commit.

**Comment by loup99** [comment]
I don't think there should be a new "imperial government". In my opinion the imperial government mod should just be used to recreate the bureaucratic government we had in CK2. The name "imperial government" should be avoided altogether, it implies all empires would have the same government form, which clearly wasn't the case.

My thoughts are that there would be some value in separating the vassals of bureaucratic realms from Sub-Roman ones, distinction that wasn't made in CK2. However, I have reservations on the implication that the themata system was already underway in 476 and I don't think this change is a priority. It could be done in a later update.

**Comment by LT-Rascek** [comment]
> I don't think there should be a new "imperial government". In my opinion the imperial government mod should just be used to recreate the bureaucratic government we had in CK2. The name "imperial government" should be avoided altogether, it implies all empires would have the same government form, which clearly wasn't the case.

Fair enough point, semantic-wise.

> My thoughts are that there would be some value in separating the vassals of bureaucratic realms from Sub-Roman ones, distinction that wasn't made in CK2. 

Okay, I can make that happen, which was part of my goal. That means I can avoid having bureaucratic vassals and use something slightly different (civil administration?) to represent the roman government at the time. It makes a code difference because certain vassal contracts should only be available to bureaucrats and not subordinates or subroman governments, so parsing them apart makes sense.

>However, I have reservations on the implication that the themata system was already underway in 476 and I don't think this change is a priority. It could be done in a later update.

Well, themata is already listed as a late migration innovation and we can pair it with military command government. Part of my concern is neither feudal nor subroman nor bureaucratic governments really capture the nature and structure of the themata system or the exarchates, which were much more akin to the military command government in the Imperial Gov't mod; the emperor had significant clout over appointments and succession was not necessarily hereditary. It's been a long-term personal annoyance of mine with CK2/CK3 to see the later Byzantine system represented as feudal. But I guess we can shelve it for now and focus on the early migration era.

> As for the distinction between military and civil, that was already in CK2, there was a Legion government.

My attempts to give legions territory in WtWSMS (and thus avoid the creation of a hereditary aristocracy in the provinces ala the Byzantine Empire) saw them degenerate to republics once landed, which defeated the purpose of landing legions in the first place. It was also an attempt to avoid having provinces inherited like feudal demenses and try and keep some measure of control over how provinces were inherited.

**Comment by LT-Rascek** [comment]
With 41a28e1, Early Migration innovations should be wrapped up as much as they can be (excepting the steppe_nomad_maa which I'll have to work on later this week); some notes:

Imperial Reconquest hasn't been revamped because:

* Regions for WRE and ERE need to be defined still in the map data
* Landed titles for WRE/ERE/Roman Empire need to be examined
* Interaction for the CB between ERE and WRE existences need to be defined (as it is in WtWSMS).

Send Missionaries character interaction has not been written in any capacity yet either, nor has  Calling Religious Councils for Late Antiquity. On the other hand, I wouldn't consider those blockers for merging this into master.

Localization and various minutia for the Late Migration and later eras will be finished sometime this week. That leaves the following on my to-do list:

* Getting Eranshar wrapped up
* Writing the Gupta government
* Writing the new succession forms (Roman Imperial and a proper form for the Sassanids instead of house seniority)
* Settling on any of the other odds and ends with respect to governments (whether or not to include military command for thematic government, the government and realm law localizations, etc.)
* And of course, the graphics are not done in any capacity (I'm not an artist by any stretch so I can only do really crude copy/paste type art).

Once those bullet points are done, I'd consider it ready for merging; precise values for various things like vassal contracts will only be determinable through play testing once the mod is in a playable state.

**Comment by loup99** [comment]
> Okay, I can make that happen, which was part of my goal. That means I can avoid having bureaucratic vassals and use something slightly different (civil administration?) to represent the roman government at the time. It makes a code difference because certain vassal contracts should only be available to bureaucrats and not subordinates or subroman governments, so parsing them apart makes sense.

"Civil administration" would be a term for a group, not a single government type. The name could be something like "gubernatorial" or "provincial".

> Well, themata is already listed as a late migration innovation and we can pair it with military command government. Part of my concern is neither feudal nor subroman nor bureaucratic governments really capture the nature and structure of the themata system or the exarchates, which were much more akin to the military command government in the Imperial Gov't mod; the emperor had significant clout over appointments and succession was not necessarily hereditary. It's been a long-term personal annoyance of mine with CK2/CK3 to see the later Byzantine system represented as feudal. But I guess we can shelve it for now and focus on the early migration era.

The thing is that already in 476 successions aren't necessarily hereditary, that isn't a difference with the themes.

> My attempts to give legions territory in WtWSMS (and thus avoid the creation of a hereditary aristocracy in the provinces ala the Byzantine Empire) saw them degenerate to republics once landed, which defeated the purpose of landing legions in the first place. It was also an attempt to avoid having provinces inherited like feudal demenses and try and keep some measure of control over how provinces were inherited.

The legions are not meant to be landed in the mod, they are attached to territories and titles but do not possess land. I don't think such drastic changes should be done at this time. So as of writing this I strongly disagree with this idea.

**Comment by LT-Rascek** [comment]
> "Civil administration" would be a term for a group, not a single government type. The name could be something like "gubernatorial" or "provincial".

Will make that change.

> The thing is that already in 476 successions aren't necessarily hereditary, that isn't a difference with the themes. Anyway, I think it is indeed important to focus firstly and then expand otherwise. With regards to inheritances, you seem to be much more concerned about succession laws than actual governments, no?

Fair enough. What we can probably do is create a succession checkbox like "forced partition" that the non-Emperor's have at start that locks the provinces to non-hereditary succession, probably at the cost of increasing the non-hereditary vassal's votes for imperial succession.

As of c5501ac, military command government references are removed.

> The legions are not meant to be landed in the mod, they are attached to territories and titles but do not possess land. I don't think such drastic changes should be done at this time. So as of writing this I strongly disagree with this idea.

Fair enough; I was more pointing out the non-hereditary succession wasn't really possible in WtWSMS than suggesting that legions be given land per se. 

As for legions in BP, creating legions should probably increase the MAAs cap at the cost of giving the legion commanders greater say in imperial succession (modeling the problems posed by giving the army too much say in succession, as was the case with the Barracks Emperors in the late 3rd century).

**Comment by LT-Rascek** [comment]
Gupta government now implemented; it's basically feudalism with access to primogeniture, since WtWSMS didn't have much flavor differentiating it from feudalism and primogeniture was used by the Guptas. 

Prototyped the new succession laws for Eranshar and Bureaucratic governments based on feudal election methodology.

Reviewing elective succession, there's no real option I see to allow _leiges_ to vote on succession (only vassals of various types as well as title holders). Until Paradox implements some equivalent to [Imperial Elective](https://ck2.paradoxwikis.com/Imperial_elective), we really can't control succession from above. So no solution on the gubernatorial front that I can see. 

On the other hand, as women can't inherit implicit claims in CK3 for male-dominated religions, we should bypass some of the problems that occurred with the common agnatic-cogantic inheritance system in Roman lands in WtWSMS.

With fe720a6, this branch is 90% done; what remains is:

* Write out the localization for governments
* Write out the localization for laws & vassal contracts
* Double check some contract rights (coinage for example) for when they should

If @loup99 @AvalonXD @herkles get a chance to start reviewing that'd be useful. I'll have to give it another review when I'm done with all the localization work.

**Comment by loup99** [comment]
We can reintroduce military command later on if we will it is needed, but for now I think it is better to focus on the actual porting of existing governments, so I salute your efforts on this. I will see if I can take the time to review it in the near future, I have to admit that I have a pretty hectic schedule at the moment.

**Comment by LT-Rascek** [comment]
> I will see if I can take the time to review it in the near future, I have to admit that I have a pretty hectic schedule at the moment. 

That's fine; when I wrap up this effort I'll move forward with #2 under the idea that most of what I've coded here for holdings/buildings is unlikely to change much at all. We've gotten most of the cultures and religions now, so the data on #3 can be put in the map data on that branch.

File: BP/common/decisions/BP_government_conversion_decisions.txt [hunk 1]
```diff
@@ -49,3 +96,290 @@ convert_to_subroman_decision = {
...
+		add_realm_law_skip_effects = bureaucratic_authority_2
+		add_realm_law_skip_effects = imperial_elective_succession_law
+	}
+
+	ai_potential = { always = yes }
+
+	ai_will_do = { base = 100 }
+}
+
+convert_to_prefeudal_decision = {
```
Comment (loup99) [inline BP/common/decisions/BP_government_conversion_decisions.txt:174]: Pre-Feudal is meant to be a dead end, you don't convert into it. If you are High Tribal the aim is to become Feudal.

File: BP/common/decisions/BP_government_conversion_decisions.txt [hunk 2]
```diff
@@ -49,3 +96,290 @@ convert_to_subroman_decision = {
...
+
+convert_to_feudal_decision = {
+	picture = "gfx/interface/illustrations/decisions/decision_realm.dds"
+	major = yes
+	desc = convert_to_feudal_decision_desc
+
+	ai_check_interval = 36
+
+	is_shown = {
+		has_government = prefeudal_government
```
Comment (loup99) [inline BP/common/decisions/BP_government_conversion_decisions.txt:241]: Pre-Feudal essentially exists in areas where feudality never developed. It is a placeholder government, but it turning into Feudal doesn't make sense.

**Comment by loup99** [comment]
A quick review since it doesn't require me to compare code: Pre-Feudal was never meant to be a transition government, it is just a grouped alternative to Feudal in areas and periods where feudalism doesn't make sense.

**Comment by LT-Rascek** [comment]
> A quick review since it doesn't require me to compare code: Pre-Feudal was never meant to be a transition government, it is just a grouped alternative to Feudal in areas and periods where feudalism doesn't make sense. In CK2 it is just a copy of Feudal, in a sense it is a dead end waiting to be removed.

Just so I'm clear: in CK3, we should remove Prefeudal entirely, or its a separate path to feudalism? I was confused because of the following:

Reviewing WtWSMS, under high tribal, the reform options listed are:
* Sub-roman, **_prefeudal_**, feudal, Eranshar, Gupta, Iqta
And for prefeudal
* Prefeudal as a reform option to feudal.

I had thought prefeudal represented the proto-feudal kingships like the Garamantes, Aksumites, Kartvelians, &c. and as such while similar to feudalism, you would have fewer options to deal with vassals:  combined tax/levy obligations making changing feudatory relationships _always_ require hooks, heavier influence of crown authority on taxes and levies, fewer legal options for things like revocation, &c.

In short, two options:

Should those groups instead be just listed as feudal? 
* The only problem I see with removing that government (in my view) is that the crown authority innovations aren't available for feudalism until the Middle Ages. 

Or should high tribals _**not**_ have access to prefeudal reform? 
* That's a simple change, but I want to be clear on your intention here; the phrase "dead end waiting to be removed" implies prefudal is deadwood in the mod _code,_ hence my confusion.

**Comment by loup99** [comment]
> Just so I'm clear: in CK3, we should remove Prefeudal entirely, or its a separate path to feudalism? I was confused because of the following:
> 
> Reviewing WtWSMS, under high tribal, the reform options listed are:
> 
> * Sub-roman, **_prefeudal_**, feudal, Eranshar, Gupta, Iqta
>   And for prefeudal
> * Prefeudal as a reform option to feudal.

In WtWSMS there is no reform attached to prefeudal. I think that is what confused you, you seem to have understood that there were.

> I had thought prefeudal represented the proto-feudal kingships like the Garamantes, Aksumites, Kartvelians, &c. and as such while similar to feudalism, you would have fewer options to deal with vassals: combined tax/levy obligations making changing feudatory relationships _always_ require hooks, heavier influence of crown authority on taxes and levies, fewer legal options for things like revocation, &c.

Yes.

> In short, two options:
> 
> Should those groups instead be just listed as feudal?
> 
> * The only problem I see with removing that government (in my view) is that the crown authority innovations aren't available for feudalism until the Middle Ages.
> 
> Or should high tribals _**not**_ have access to prefeudal reform?
> 
> * That's a simple change, but I want to be clear on your intention here; the phrase "dead end waiting to be removed" implies prefudal is deadwood in the mod _code,_ hence my confusion.

Sorry if I didn't express myself clearly. It is the second option that I meant, prefeudal is "dead" in the sense that it isn't included in the reform paths and would ideally be removed by regionalised unique governments. For now it stays since we don't have those unique regional governments.

**Comment by LT-Rascek** [comment]
> In CK2 there is no reform attached to prefeudal. I think that is what confused you, you seem to have understood that there were, when it isn't an option.

> Sorry if I didn't express myself clearly. It is the second option that I meant, prefeudal is "dead" in the sense that it isn't included in the reform paths and would ideally be removed by regionalised unique governments. For now it stays since we don't have those unique regional governments.

Okay makes sense.

**Comment by loup99** [comment]
Can I set this pull request as "ready for review" or do you want it to remain a draft?

**Comment by LT-Rascek** [comment]
> Can I set this pull request as "ready for review" or do you want it to remain a draft?

Draft still; I have the following points left to-do:

* Remove "convert to prefeudal decision"
* Write out the localization for governments
* Write out the localization for laws & vassal contracts
* Double check some contract rights (coinage for example) for when they should

**Comment by LT-Rascek** [comment]
Rebased onto ad54518 and cleaned up for review.

**Comment by loup99** [comment]
> Rebased onto [ad54518](https://github.com/loup99/BP/commit/ad54518fa0231937972a5fd2bc4215d1629c026a) and cleaned up for review.
> Will note that graphics are not implemented as I wasn't sure which graphics we wanted to use for things like eras.
> 
> Most of the issues I find now are just minor things like missing localizations and graphics. The code base in this PR is now at the point where I can reasonably interact with those debug decisions I wrote and start hammering the localization issues now.

Ok, there are over 60 files modified in the pull request, is there anything you want me to review in priority ?

File: BP/common/casus_belli_types/00_dejure_war.txt [hunk 3]
```diff
@@ -0,0 +1,1758 @@
...
+	cb_name_no_target = DEJURE_CB_DUCHY_NAME_GENERIC
+	interface_priority = 99 # Below the "take all" one
+	
+	attacker_wargoal_percentage = 0.8
+	
+	max_ai_diplo_distance_to_title = 500
+	min_ai_score = 100
+}
+
+imperial_reconquest_cb = { #TODO: Rework in context of E/W Roman Empire and Roman Successors
```
Comment (loup99) [inline BP/common/casus_belli_types/00_dejure_war.txt:1268]: We already reworked this CB in CK2.

File: BP/common/casus_belli_types/00_religious_war.txt [hunk 4]
```diff
@@ -0,0 +1,3717 @@
...
+
+	use_de_jure_wargoal_only = yes
+
+	attacker_wargoal_percentage = 0.8
+
+	max_ai_diplo_distance_to_title = 500
+	min_ai_score = 100
+}
+
+religious_war = {
```
Comment (loup99) [inline BP/common/casus_belli_types/00_religious_war.txt:436]: Holy War should be disabled like in CK2, but what is the point of the changes you made in this file?

File: BP/localization/english/BP_decisions_l_english.yml [hunk 5]
```diff
@@ -4,3 +4,35 @@
...
+ convert_to_high_tribal_decision_confirm:0 "Confirm"
+ 
+ convert_to_subroman_decision:0 "Convert to Subroman Government"
+ convert_to_subroman_decision_desc:0 "We abandon our tribal ways and embrace, to the best of our ability, the ways of our Roman ancestors."
+ convert_to_subroman_decision_tooltip:0 "Change [government|E] type by becoming a Sub-Roman government"
+ convert_to_subroman_decision_confirm:0 "Adopt the Faded Roman Institutions"
+ 
+ convert_to_bureaucratic_decision:0 "Convert to Bureacratic Government"
+ convert_to_bureaucratic_decision_desc:0 "Descended from the great Romans, we will embrace and reinvigorated their ways, adopting the customs, mannerism, and goverance of Imperial Rome."
+ convert_to_bureaucratic_decision_tooltip:0 "Change [government|E] type by becoming a Sub-Roman government"
```
Comment (loup99) [inline BP/localization/english/BP_decisions_l_english.yml:26]: I think this is a mistake.

File: BP/common/decisions/BP_government_conversion_decisions.txt [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+	major = yes
+	desc = convert_to_feudalism_decision_desc
+
+	ai_check_interval = 36
+
+	is_shown = {
+		OR = {
+			has_government = prefeudal_government
+			has_government = high_tribal_government 
+			has_government = subroman_government
```
Comment (loup99) [inline BP/common/decisions/BP_government_conversion_decisions.txt:213]: Sub-Roman aren't enable to convert to Feudal through decision

File: BP/common/decisions/BP_government_conversion_decisions.txt [hunk 7]
```diff
@@ -0,0 +1,402 @@
...
+	major = yes
+	desc = convert_to_clan_decision_desc
+
+	ai_check_interval = 36
+
+	is_shown = {
+		OR = {
+			has_government = prefeudal_government
+			has_government = high_tribal_government
+			has_government = subroman_government
```
Comment (loup99) [inline BP/common/decisions/BP_government_conversion_decisions.txt:314]: Same as above.

File: BP/common/casus_belli_types/00_dejure_war.txt (hunk 3 above)
Comment (LT-Rascek) [reply BP/common/casus_belli_types/00_dejure_war.txt:1268]: Correct; I haven't ported it in yet because I haven't defined all the new regions on the map that go with it. Reason:  that seemed to require refactoring de jure titles first before fully implementing.

File: BP/common/casus_belli_types/00_religious_war.txt (hunk 4 above)
Comment (LT-Rascek) [reply BP/common/casus_belli_types/00_religious_war.txt:436]: It was; I noted in my institution proposal to gate holy wars by innovations:

* Early Migration Era: Doctrine and Dogma -> County Holy War
* Late Migration Era: Spiritual Canon -> Duchy Holy War
* Late Migration Era: Divine Mandate (Central Semitic Cultures) -> All Holy Wars
* Early Medevial Era: Established Faith -> Kingdom Holy War

Alternatively, we can completely block those with Established Faith, but I thought a progressive increase in scale seemed more logical. Similarly, the Arab cultures get to unlock all of the Holy Wars earlier to help model the Caliphates early expansion.

File: BP/localization/english/BP_decisions_l_english.yml (hunk 5 above)
Comment (LT-Rascek) [reply BP/localization/english/BP_decisions_l_english.yml:26]: Yeah, that's a typo that needs to be fixed.

File: BP/common/decisions/BP_government_conversion_decisions.txt (hunk 6 above)
Comment (LT-Rascek) [reply BP/common/decisions/BP_government_conversion_decisions.txt:213]: Will do.

File: BP/common/decisions/BP_government_conversion_decisions.txt (hunk 7 above)
Comment (LT-Rascek) [reply BP/common/decisions/BP_government_conversion_decisions.txt:314]: Will do.

**Comment by LT-Rascek** [comment]
> Ok, there are over 60 files modified in the pull request, is there anything you want me to review in priority ?

Nothing in particular; this branch touches a lot of the code so most of it will require review.

File: BP/common/casus_belli_types/00_dejure_war.txt (hunk 3 above)
Comment (loup99) [reply BP/common/casus_belli_types/00_dejure_war.txt:1268]: Yes, no problems, I just wanted to point out there was a skeleton. 😉

File: BP/common/casus_belli_types/00_religious_war.txt (hunk 4 above)
Comment (loup99) [reply BP/common/casus_belli_types/00_religious_war.txt:436]: The progressive scaling makes sense but I think it should be delayed in time, so County Holy War happens much later than the Early Migrational Era.

Comment (LT-Rascek) [reply BP/common/casus_belli_types/00_religious_war.txt:436]: County Holy Wars in the Late Migration, Duchy/Kingdom Holy Wars in the Early Medevial Era then, with Central Semetics and Muslim always having access to all holy wars?

Comment (loup99) [reply BP/common/casus_belli_types/00_religious_war.txt:436]: Doesn't the Rise of Islam happen by late Late Migration? I would disagree with Central Semetics always having access to all holy wars, to me that feels like railroading.

Comment (LT-Rascek) [reply BP/common/casus_belli_types/00_religious_war.txt:436]: Removed divine mandate and instead islam religion always has access to all holy wars. Furthermore, county holy war is now only available in late migration era and duchy/kingdom holy wars in early migration era.

**Comment by LT-Rascek** [comment]
Rebased onto master; Islam now can use all holy war types while other religions are gated behind innovations. County holy war is unlockable in the late migration era, while duchy and kingdom holy war is unlockable in the early medieval era.

That resolves your previous concerns. If you have other concerns, let me know.

**Comment by loup99** [comment]
What are you modifying in the buildings files? I only find vanilla updates which haven't been ported, no content relating to the mod, but I might be mistaken

File: BP/common/casus_belli_types/00_dejure_war.txt [hunk 8]
```diff
@@ -0,0 +1,1758 @@
+﻿de_jure_cb = {
```
Comment (loup99) [inline BP/common/casus_belli_types/00_dejure_war.txt:1]: There is some updating left to do in this file.

File: BP/common/casus_belli_types/00_religious_war.txt [hunk 9]
```diff
@@ -0,0 +1,3717 @@
+﻿minor_religious_war = {
```
Comment (loup99) [inline BP/common/casus_belli_types/00_religious_war.txt:1]: Same as the other file, it needs to be merged with vanilla.

File: BP/common/culture/innovations/00_cultural_maa_innovations.txt [hunk 10]
```diff
@@ -0,0 +1,631 @@
+﻿#### Local Types ####
```
Comment (loup99) [inline BP/common/culture/innovations/00_cultural_maa_innovations.txt:1]: This file should also be updated from vanilla

File: BP/common/culture/innovations/00_early_medieval_innovations.txt [hunk 11]
```diff
@@ -0,0 +1,265 @@
+﻿#### Local Types ####
```
Comment (loup99) [inline BP/common/culture/innovations/00_early_medieval_innovations.txt:1]: Same as above.

**Review by loup99 — COMMENTED** [review]
There are a few files which are not up to date to vanilla, which makes it difficult to review. I have commented on some of them but it is not exhaustive. I think you should go through the files of the pull request in general to compare with vanilla updates.

**Comment by LT-Rascek** [comment]
> What are you modifying in some of the buildings files, like 00_duchy_capital_buildings.txt? I mostly find vanilla updates which haven't been ported, not always content relating to the mod, but I might be mistaken. Regardless there are clearly conflicts with the vanilla update, so there is some merging left to do there to clean up.

Will do sometime this week. Probably had an issue porting between 1.2 and 1.3 that missed some things.

**Comment by loup99** [comment]
> > What are you modifying in some of the buildings files, like 00_duchy_capital_buildings.txt? I mostly find vanilla updates which haven't been ported, not always content relating to the mod, but I might be mistaken. Regardless there are clearly conflicts with the vanilla update, so there is some merging left to do there to clean up.
> 
> Will do sometime this week. Probably had an issue porting between 1.2 and 1.3 that missed some things.

Ok, no problems.

**Comment by loup99** [comment]
> > What are you modifying in some of the buildings files, like 00_duchy_capital_buildings.txt? I mostly find vanilla updates which haven't been ported, not always content relating to the mod, but I might be mistaken. Regardless there are clearly conflicts with the vanilla update, so there is some merging left to do there to clean up.
> 
> Will do sometime this week. Probably had an issue porting between 1.2 and 1.3 that missed some things.

If you prefer I can do this myself otherwise, if you want to focus on the other pull request with the titles for now.

**Comment by LT-Rascek** [comment]
> If you prefer I can do this myself otherwise, if you want to focus on the other pull request with the titles for now.

With 2b47129 on the titles branch, I finished the historical Roman Empire provinces and was about to shift gears to this. Although, I prefer doing more map work, TBH, because I've gotten to a good groove with doing that work.

But I can wrap this up; the titles' branch is now at (what is for me) a good stopping point and I was going to take a break for a bit before doing more implementation work regardless.

**Comment by loup99** [comment]
> > If you prefer I can do this myself otherwise, if you want to focus on the other pull request with the titles for now.
> 
> With [2b47129](https://github.com/loup99/BP/commit/2b471296cd742500ad4fc8de3824d04c47caab8c) on the titles branch, I finished the historical Roman Empire provinces and was about to shift gears to this. Although, I prefer doing more map work, TBH, because I've gotten to a good groove with doing that work.
> 
> But I can wrap this up; the titles' branch is now at (what is for me) a good stopping point and I was going to take a break for a bit before doing more implementation work regardless.

Ok, I just don't want you to feel like you are overwhelmed with work, especially if you want to take a break.

**Comment by LT-Rascek** [comment]
> Ok, I just don't want you to feel like you are overwhelmed with work, especially if you want to take a break, which is entirely fair.

Not overwhelmed nor offended if people want to take it off my hands. I just tend not to multitask; I prefer to get whatever task to a good stopping point before switching gears.

**Comment by LT-Rascek** [comment]
@loup99 955e39c should fix all the remaining issues from Northern Lords, rebasing onto master, and removing some deadwood code or unmodified vanilla files. Let me know if you have other questions or concerns.

**Comment by loup99** [comment]
In some files there will have to be additional cleaning up later on, I think it would be better to add our new buildings in separate files to avoid confusion when merging, but the pull request is otherwise ready to be merged.
